pkg/controllers/gateway_controller.go

```diff
@@ -224,18 +224,35 @@ func (r *gatewayReconciler) reconcileUpsert(ctx context.Context, gw *gwv1beta1.G
 		return err
 	}
 
-	snInfo, err := r.cloud.Lattice().FindServiceNetwork(ctx, gw.Name)
+	foundSnArns := make([]string, 0, 1)
+	serviceNetworks, err := r.cloud.Lattice().ListServiceNetworksWithContext(ctx, &vpclattice.ListServiceNetworksInput{})
 	if err != nil {
-		if services.IsNotFoundError(err) {
-			if err = r.updateGatewayProgrammedStatus(ctx, "", gw, false); err != nil {
-				return lattice_runtime.NewRetryError()
-			}
-			return nil
-		}
 		return err
 	}
+	for _, sn := range serviceNetworks.Items {
+		if *sn.Id == gw.Name {
+			foundSnArns = []string{*sn.Arn}
+			break
+		} else if *sn.Name == gw.Name {
+			foundSnArns = append(foundSnArns, *sn.Arn)
+		}
+	}
+
+	if len(foundSnArns) == 0 {
+		if err = r.updateGatewayProgrammedStatus(ctx, gw, gwv1.GatewayReasonPending, "VPC Lattice Service Network not found"); err != nil {
+			return lattice_runtime.NewRetryError()
+		}
+		return nil
+	}
+
+	if len(foundSnArns) > 1 {
+		if err = r.updateGatewayProgrammedStatus(ctx, gw, core.GatewayReasonConflicted, "Multiple service networks with the same name were found. Either ensure only one service network with the same name is visible in the AWS account or use the desired service network's id as the Gateway name."); err != nil {
+			return lattice_runtime.NewRetryError()
+		}
+		return nil
+	}
```

Comment:
- status updates for Accepted and Programmed should be in single API call, update both statuses and then call k8s-api
- group validation logic into own method
- UpdateGWListenerStatus already doing some job, need to refactor. Structure should be UpdateStatus -> UpdateAccepted, UpdateProgrammed -> Call K8s API with all statuses populated

Reply:
Will change back to old error checking logic and add catch for name conflicts so that we can set the programmed status to Invalid instead of Conflicted